ui/lib/kv/addon/routes/secret.js

```diff
@@ -32,4 +33,15 @@ export default class KvSecretRoute extends Route {
       metadata: this.fetchSecretMetadata(backend, path),
     });
   }
+
+  @action
+  willTransition(transition) {
+    // refresh the route if transitioning to secret.index (which happens after delete, undelete or destroy)
+    // or transitioning from editing either metadata or secret data (creating a new version)
+    const performedToolbarAction = transition.to.name === 'vault.cluster.secrets.backend.kv.secret.index';
+    const afterEditing = transition.from.localName === 'edit';
+    if (performedToolbarAction || afterEditing) {
+      this.refresh();
+    }
+  }
```

Comment:
I previously had the conditional above as the following - which might be easier to read? wanted it to be clear what was happening here
```js
const { to, from } = transition;
if (to.name === 'vault.cluster.secrets.backend.kv.secret.index' || from.localName === 'edit') 
```